### ISSUE TITLE
stack new test yesod-postgres creates invalid .cabal file

### ISSUE THREAD

**Issue by hyperpape** [yesodweb/yesod-scaffold#178]
Running 
      
     stack new test yesod-postgres
     cd test && stack build

which fails with: 

```
Building all executables for `test' once. After a successful build of all of them, only specified executables will be rebuilt.
test-0.0.0: configure (lib + exe)
Configuring test-0.0.0...
There exist several components with the same name: 'test'. This is a bug in your .cabal file.
CallStack (from HasCallStack):
  error, called at libraries/Cabal/Cabal/Distribution/PackageDescription/Utils.hs:18:15 in Cabal-2.0.1.0:Distribution.PackageDescription.Utils
```

`stack new test` produces a project that can be built.

**Comment by snoyberg** [comment]
I'm OK with changing the test suite name to `PROJECTNAME-test`. Would you be interested in sending a PR?

**Comment by hyperpape** [comment]
Sounds good, I can probably do a PR.

**Comment by snoyberg** [comment]
Fixed by #181
